Automattic/jetpack
# Protect: Hide WAF screen on unsupported platforms

This PR disables the WAF screen from being rendered and included in navigation when the `DISABLE_JETPACK_WAF` constant is set. This is primarily to prevent the settings from being accessible on Atomic sites.

#### Changes proposed in this Pull Request:

* Extracts `DISABLE_JETPACK_WAF` killswitch logic into `Waf_Runner` so that is can be used outside of the package.
* Passes the killswitch value into the Protect app via a `isSupported` initial state variable.
* Does not render the Firewall tab in the navigation when `! isSupported`.
* Redirects away from the `/firewall` page to `/` when `! isSupported`.

#### Other information:


#### Jetpack product discussion



1201069996155224-as-1203527484103970

#### Does this pull request change what data or activity we track or use?



No

#### Testing instructions:






* Validate the firewall tab is accessible, and firewall page operates normally.
* Add the `DISABLE_JETPACK_WAF` constant to your environment.
* Validate the firewall tab is no longer rendered or accessible.
* Remove the constant.
* Validate the firewall tab is accessible, and firewall page operates normally.
